src/main/java/logging/Logger.java

```diff
@@ -109,37 +111,38 @@ private synchronized void log(LogLevel level, String text)
 
     /**
      * Logs a byte array to verbose output
-     * @param bytes The byte array to log
+     * @param packet The packet to log
      */
-    public synchronized void logVerbose(byte[] bytes)
+    public synchronized void logVerbose(DatagramPacket packet)
     {
-        logVerbose(getByteArrayString(bytes));
+        logVerbose(getByteArrayString(packet));
     }
 
     /**
      * Logs a byte array to quiet output
-     * @param bytes The byte array to log
+     * @param packet The packet to log
      */
-    public synchronized void logQuiet(byte[] bytes)
+    public synchronized void logQuiet(DatagramPacket packet)
     {
-        logQuiet(getByteArrayString(bytes));
+        logQuiet(getByteArrayString(packet));
     }
 
     /**
-     * @param bytes The byte array to print
+     * @param packet The packet to print
      * @return A comma separated list of all bytes in the array
      */
-    private synchronized String getByteArrayString(byte[] bytes)
+    private synchronized String getByteArrayString(DatagramPacket packet)
```

Comment:
Should this method name be updated or is this fine? It still outputs a byte array as a string.

Reply:
IMO I think this should actually be kept with a byte array, but adding int offset and int length as parameters so that we can print truncated byte arrays. I think its reasonable to do this because we can print datagram packets by doing the following:

getByteArrayString(packet.getData(), 0, packet.getLength());


This way we only print the bytes that are necessary